src/visualizers/P5Visualizer.ts

```diff
@@ -0,0 +1,171 @@
+import type {VisualizerInterface} from './VisualizerInterface'
+import {Paramable} from '../shared/Paramable'
+import type {SequenceInterface} from '../sequences/SequenceInterface'
+import {CachingError} from '../sequences/Cached'
+import p5 from 'p5'
+
+// Ugh, the gyrations we go through to keep TypeScript happy
+// while only listing the p5 methods once:
+
+class WithP5 extends Paramable {
+    deviceMoved() {}
+    deviceShaken() {}
+    deviceTurned() {}
+    doubleClicked() {}
+    keyPressed() {}
+    keyReleased() {}
+    keyTyped() {}
+    mouseClicked() {}
+    mouseDragged() {}
+    mouseMoved() {}
+    mousePressed() {}
+    mouseReleased() {}
+    mouseWheel() {}
+    setup() {}
+    touchEnded() {}
+    touchMoved() {}
+    touchStarted() {}
+    windowResized() {}
```

Comment:
I'm curious where you found this list of possible events.  Does p5 provide it somewhere?  It's all mixed in with everything else in the reference pages.

Reply:
I just went to https://p5js.org/reference/, and looked in the Events and Environment sections, and looked for all of the places where it seemed that p5 automagically calls a function that may be present on the sketch object, and listed them all here. It is totally possible that I may have missed such a call or that there might sneakily be such calls in other sections of the p5 reference, but I figured if anyone in the future wanted p5 to be calling some other method (or if a future release of p5 added more) we would just do a PR that adds to this list.

Reply:
sounds good, resolving